elasticapm/transport/base.py

```diff
@@ -100,28 +98,28 @@ def flush(self, sync=False, start_flush_timer=True):
         :param start_flush_timer: set to True if the flush timer thread should be restarted at the end of the flush
         :return: None
         """
+        self._stop_flush_timer()
         with self._queue_lock:
-            self._stop_flush_timer()
             queued_data, self._queued_data = self._queued_data, None
-            if queued_data and not self.state.should_try():
-                logger.error("dropping flushed data due to transport failure back-off")
-            elif queued_data:
-                fileobj = queued_data.fileobj  # get a reference to the fileobj before closing the gzip file
-                queued_data.close()
-
-                # StringIO on Python 2 does not have getbuffer, so we need to fall back to getvalue
-                data = fileobj.getbuffer() if hasattr(fileobj, "getbuffer") else fileobj.getvalue()
-                if hasattr(self, "send_async") and not sync:
-                    self.send_async(data)
-                else:
-                    try:
-                        self.send(data)
-                        self.handle_transport_success()
-                    except Exception as e:
-                        self.handle_transport_fail(e)
-            self._last_flush = timeit.default_timer()
-            if start_flush_timer:
-                self._start_flush_timer()
+        if queued_data and not self.state.should_try():
+            logger.error("dropping flushed data due to transport failure back-off")
```

Comment:
to be clear, this is the important line to move outside?

Reply:
that one, and the two `Starting/Cancelling flush timer`